bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/api/SiteApi.java

```diff
@@ -0,0 +1,266 @@
+            final ISiteResponseListener siteResponseListener, final String pollId) {
+        if (requestLimiter.getRequestCountIfAvailable() == RequestLimiter.INVALID_REQUEST_ID) {
+            logger.debug("{} - Disabled requesting data - request limit has been hit", usageId);
+            return false;
+        }
+
+        notifyRateLimiterListeners();
+
+        final Response.CompleteListener listener = new BufferingResponseListener() { // 4.5kb buffer will cover both
+            @Override
+            public void onComplete(@Nullable Result result) {
+                if (result != null) {
+                    final boolean userAuthValidatedPreviously = apiAuth.getIsAuthenticated();
+                    apiAuth.processResult(result);
+                    if (!apiAuth.getIsAuthenticated()) {
+                        // Callback Async function confirming authorization is completed.
+                        notifyAuthenticationListeners(false);
+                        return;
+                    }
+
+                    // User is authorized at this point
+                    if (!userAuthValidatedPreviously) {
+                        // Authorization token confirmed
+                        // Callback Async function confirming authorization is completed.
+                        notifyAuthenticationListeners(true);
+                    }
+
+                    if (result.isSucceeded()) {
+                        final String response = getContentAsString();
+                        if (response != null) {
+                            logger.trace("Got response for poll ID: \"{}\"", pollId);
+                            notifyConnectedListeners();
+                            scheduler.execute(() -> {
+                                if (daily) {
+                                    siteResponseListener.processDailyResponse(response, pollId);
+                                } else {
+                                    siteResponseListener.processHourlyResponse(response, pollId);
+                                }
+                            });
+                        }
+                    } else {
+                        notifyCommFailureListeners(result.getFailure());
+                    }
+                }
+            }
+        };
+
+        sendAsyncSiteApiRequest(daily, location, listener, pollId);
+        return true;
+    }
+
+    public void sendAsyncSiteApiRequest(final boolean daily, final PointType location,
+            final Response.CompleteListener listener, final String pollId) {
+        final String url = ((daily) ? GET_FORECAST_URL_DAILY : GET_FORECAST_URL_HOURLY)
+                .replace(GET_FORECAST_KEY_LATITUDE, location.getLatitude().toString())
+                .replace(GET_FORECAST_KEY_LONGITUDE, location.getLongitude().toString());
+
+        final Request request = httpClientFactory.getCommonHttpClient().newRequest(url).method(HttpMethod.GET)
```

Comment:
When the constructor is called you could (and should) cache the httpclient and not the factory. This would prevent to create a httpclient for each call.

Reply:
Yeah I cant remember why I didn't do that originally, I think during the original refactors it may not have fitted. I've adjusted definitely now to how you suggested thank you :)